treeherder/model/models.py

```diff
@@ -941,45 +941,11 @@ def best_automatic_match(self, min_score=0):
                             .select_related('classified_failure')
                             .first())
 
-    def set_classification(self, matcher_name, classification=None, bug_number=None,
-                           mark_best=False):
-        with transaction.atomic():
-            if classification is None:
-                if bug_number:
-                    classification, _ = ClassifiedFailure.objects.get_or_create(
-                        bug_number=bug_number)
-                else:
-                    classification = ClassifiedFailure.objects.create()
-
-            new_link = FailureMatch(
```

Comment:
The other `set_classification` implementation creates a new `TextLogErrorMatch` link rather than `FailureMatch`. Is this equivalent? I really don't know enough about this code to say (or have the time to dig in right now), so happy to take your word for it :-)

Reply:
Yes and no… (sorry ;).

Before this change we created `TextLogErrorMatch` and `FailureMatch` instances everywhere we saved a Match to the DB.  The `set_classification` methods had to account for this on both models.  This change consolidates to one Match object (`TextLogErrorMatch`) so the call sites of this particular function now point to the relevant `TextLogError`.

Reply:
👍